### PR TITLE
feat: identify TX stake credential

### PR DESCRIPTION
**Pull request by cong-or**
Identify TX stake credential from certificates and witness metadata.

### CODE REVIEW

File: catalyst-gateway/bin/src/util.rs [hunk 1]
```diff
@@ -70,3 +80,82 @@ pub fn parse_child_assets(assets: &[MultiEraAsset]) -> Vec<Asset> {
         })
         .collect()
 }
+
+/// Eras before staking should be ignored
+pub fn valid_era(era: Era) -> bool {
+    matches!(
+        era,
+        Era::Shelley | Era::Allegra | Era::Mary | Era::Alonzo | Era::Babbage | Era::Conway
+    )
```
Comment (FelipeRosa) [inline catalyst-gateway/bin/src/util.rs:89]: Maybe change this to something like:

```suggestion
    !matches!(era, Era::Byron)
```

So when future eras come we don't ignore them?

File: catalyst-gateway/bin/src/event_db/utxo.rs [hunk 2]
```diff
@@ -45,9 +74,11 @@ impl EventDB {
                                         .to_string(),
                                 )
                             })?,
-                            &tx.hash().as_slice(), /* temporary until we have foreign key
-                                                    * relationship in the context of
-                                                    * registrations */
+                            &hex::decode(stake_credential_hash.clone()).map_err(|e| {
```
Comment (Mr-Leshiy) [inline catalyst-gateway/bin/src/event_db/utxo.rs:77]: Is it possible to take just a reference at this place ?
```Rust
 &hex::decode(&stake_credential_hash)
 ```

File: catalyst-gateway/bin/src/util.rs (hunk 1 above)
Comment (cong-or) [reply catalyst-gateway/bin/src/util.rs]: yea we're just ignoring byron for now.

File: catalyst-gateway/bin/src/event_db/utxo.rs (hunk 2 above)
Comment (cong-or) [reply catalyst-gateway/bin/src/event_db/utxo.rs]: yes just changed it 👍

**Review by Mr-Leshiy — APPROVED** [review]
LGTM

**Pull request merged**